### ISSUE TITLE
perf: Overflow Scroll Memory Leak

### ISSUE THREAD

**Issue by iamchairs** [driftyco/ionic#4710]
**Type**: <span ionic-type>perf</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>When added `overflow-scroll` to an `ion-content` element there is a memory leak issue. Both in events as well as DOM.

![Memory Leak](http://s11.postimg.org/bals8v1gj/ontapoverflowscroll.png)

Here is a codepen that reproduces the problem.

http://codepen.io/anon/pen/VewYmO

Cycle between `Page 1` and `Page 2` and watch your memory soar.</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @iamchairs!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=4710). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=4710)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by mlynch** [comment]
Thanks for the report, we will look into this.

**Comment by mlynch** [comment]
Hey @iamchairs, take a look at this forked codepen: http://codepen.io/anon/pen/OMJgZG

The all to arr() in the template caused angular to repeatedly call that function. With a change to doing it once in the page controller, memory and performance is a lot better (which is to be expected).

Let us know if you're still seeing an issue beyond that.

**Comment by iamchairs** [comment]
Yes, that is more performent . But my goal with the codepen wasn't to create the most performent ng-repeat, it was to illustrate the memory leak.

Which still exists.

![Still leaking](http://s29.postimg.org/40mivvi47/stillleaking.png)

**Comment by andreialecu** [comment]
This may be related to Hammer/Gesture listeners. I'm seeing some references to "bindDomOnTouch" being retained. Possibly related: https://github.com/hammerjs/hammer.js/issues/353

**Comment by mlynch** [comment]
Thanks, we are tracking this and will if we can get to the bottom of it.

**Comment by iamchairs** [comment]
If it helps, you don't even need to set the directive. If you turn off jsScrolling in the ionic config it has the same result.

    $ionicConfigProvider.scrolling.jsScrolling(false);

http://codepen.io/anon/pen/wMBzEW

**Comment by iamchairs** [comment]
PR up. If we could get this in sooner than 1.3 that would be great.

**Comment by mlynch** [comment]
Thanks so much! Merged the PR and this will be in 1.2 hopefully tomorrow or wednesday if all goes well.

**Comment by dioxmio** [comment]
I have tried with 1.2.4 and I can see that when I use `overflow-scroll="true"` some nodes get lost whereas if I remove it no nodes are lost (I'm just switching over two screen, super simple scenario). I have tried v1.2.4-nightly-2998 and whereas it seems to be a bit better the leak is still there. I would like to reduce as much as possible since we developing for mobile.

In this two screenshots I'm just switching overflow to true/false

![screen shot 2016-04-16 at 12 19 37](https://cloud.githubusercontent.com/assets/3042877/14580453/ac298c30-03cd-11e6-89b1-c73f4e10fb3f.png)
![screen shot 2016-04-16 at 12 18 04](https://cloud.githubusercontent.com/assets/3042877/14580454/ac2abb82-03cd-11e6-9340-53f0f645dbe3.png)

many thanks,

jose
